volatility3/framework/symbols/windows/extensions/__init__.py

```diff
@@ -492,9 +492,39 @@ def get_name(self) -> str:
         return header.NameInfo.Name.String  # type: ignore
 
 
-class ETHREAD(objects.StructType):
+class ETHREAD(objects.StructType, pool.ExecutiveObject):
     """A class for executive thread objects."""
 
+    def is_valid(self) -> bool:
+        """Determine if the object is valid."""
+
+        try:
+            
+            # validation by TID:
+            if self.Cid.UniqueThread % 4 != 0: # NT tids are divisible by 4
+                return False
+            
+            # validation by PID of parent process:
+            if self.Cid.UniqueProcess % 4 != 0:
+                return False
+            
+            # validation by thread creation time:
+            if self.Cid.UniqueProcess != 4:     # The System process (PID 4) has no create time
+                ctime = self.get_create_time()
+                if not isinstance(ctime, datetime.datetime):
+                    return False
+
+            # passed all valitations
+            return True
+        except:
+            return False
```

Comment:
regarding this -
I completely understand your concern. In general, you are absolutely right, and it's a good practice.
However, in this specific case, I noticed that other classes' implementations of is_valid do not log the exceptions, and that's why I did not log it either. In particular, I found this pattern in the EPROCESS and CONTROL_AREAD classes.
Based on my understanding, exceptions that occur in that clause indicate that the object that's being checked is not a proper `_Ethread` kernel struct, but random data found after a byte sequence matching the pool tag. Considering the expected frequency of these exceptions, I decided to comply with the implementations of `is_valid` in the other two classes.
If you think that would be beneficial, I will of course add logging to the exception handler.
Please just let me know your preference (:

Reply:
Yep, that's fair enough, I didn't spot the function it was in, and you're right, if someone's testing for validity, it's not critical they're told why it failed since they should end up still handling the issue gracefully.  Happy to mark this as resolved.